From d27cc3568c2c04e86a8ec6e29fcdf7e3814b0596 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Fri, 15 May 2020 09:25:40 +0200
Subject: [PATCH 1/3] use SetFilePointerEx instead of SetFilePointer

It's available on more Win10 versions with UCRT builds and provides the same
features. The API is available since Windows XP.
---
 taglib/toolkit/tfilestream.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/taglib/toolkit/tfilestream.cpp b/taglib/toolkit/tfilestream.cpp
index 5205bae0..4b448271 100644
--- a/taglib/toolkit/tfilestream.cpp
+++ b/taglib/toolkit/tfilestream.cpp
@@ -364,7 +364,9 @@ void FileStream::seek(long offset, Position p)
   }
 
   SetLastError(NO_ERROR);
-  SetFilePointer(d->file, offset, NULL, whence);
+  LARGE_INTEGER lOffset;
+  lOffset.QuadPart = offset;
+  SetFilePointerEx(d->file, lOffset, NULL, whence);
 
   const int lastError = GetLastError();
   if(lastError != NO_ERROR && lastError != ERROR_NEGATIVE_SEEK)
@@ -411,7 +413,9 @@ long FileStream::tell() const
 #ifdef _WIN32
 
   SetLastError(NO_ERROR);
-  const DWORD position = SetFilePointer(d->file, 0, NULL, FILE_CURRENT);
+  LARGE_INTEGER lOffset;
+  lOffset.QuadPart = 0;
+  const DWORD position = SetFilePointerEx(d->file, lOffset, NULL, FILE_CURRENT);
   if(GetLastError() == NO_ERROR) {
     return static_cast<long>(position);
   }
-- 
2.26.0.windows.1

